_docs/for-contributors/2017-01-01-building-from-source.md

```diff
@@ -205,3 +205,113 @@ Now, to run the wallet connected to the Cardano SL in dev-mode, call:
 ```
 CARDANO_API=true npm run dev
 ```
+
+## Building Cardano SL node for Testing Purpose
+
+If you want to test functionality of Cardano SL node in our [testnet](/timeline/testnet/), you should go to the root directory of the repository and switch to release branch, for example, `cardano-sl-0.2`. After that build a node with this command:
+
+~~~ bash
+CSL_SYSTEM_TAG=linux64 stack build --flag cardano-sl:-dev-mode
+~~~
+
+It means that you build a node in **production** mode.
+
+Now run your node with this command:
+
+~~~ bash
+stack exec -- cardano-node \
+    --db-path node-db0 \
+    --rebuild-db \
+    --dht-key a_P8zb6fNP7I2H54FtGuhqxaMDAwMDAwMDAwMDAwMDA= \
+    --listen 127.0.0.1:3000 \
+    --json-log logs/2017-03-10_133948/node0.json \
+    --logs-prefix logs/2017-03-10_133948 \
+    --log-config logs.yaml \
+    --kademlia-dump-path logs/2017-03-10_133948/dump/kademlia0.dump \
+    --wallet \
+    --keyfile secret.key \
+    --peer 35.156.234.240:3000/S-GJ1HDZyHpIS_ZOFBlZsipkGNXy-tmCDtY_CvaiywE= \
+    --peer 35.157.148.244:3000/nzcEO6Cv7tKD3REWoWc9vcHMOZEQWFGLX4150bt_cYw= \
+    --peer 52.52.249.66:3000/dYGuDj0BrJxCsTC9ntJE7ePT7wUoVdQMH3sKLzQD8bo= \
+    --peer 52.8.23.163:3000/8hVd3J5Jc0GCkIV-QeiOkwDqH38QooYGo3iHJnZuIk4=
+~~~
+
+You can download `secret.key` from [Cardano Test-Net Faucet](https://tada.iohk.io).
+
+After running with this command your node will listen two ports:
+
+1. `3000` for communication with other nodes in our testnet,
+2. `8090` for use wallet functionality, via wallet web API.
+
+Now you can test your node.
+
+## Use Case - Sending Transaction
+
+Although you can work with a node via Daedalus GUI, the simplest way to test a node is a CLI-based way. The main purpose of a node is a sending/receiving money, so let's create your first wallet and initiate your first transaction.
+
+You can create new wallet with this command:
+
+~~~ bash
+curl -X POST -H "Content-Type: application/json" -d '{"cwBackupPhrase": {"bpToList": ["squirrel", "material", "silly", "twice", "direct", "slush", "pistol","razor", "become", "junk", "kingdom", "flee"]},"cwInitMeta": {"cwType": "CWTPersonal","cwCurrency": "ADA","cwName": "Personal Wallet 1"}}' http://localhost:8090/api/new_wallet
```

Comment:
This is excellent, having a sample of how to poke the endpoints with `curl`. 
However, speaking as someone who spent quite some time just trying to get the commands right (this one in particular), this section needs to be clarified in the sense that to each backup phrase corresponds a single secret key - trying to copy the phrase from the example will always replicate the same key.